### ISSUE TITLE
Inconsistent implicit long-literal-to-double cast in function params (32/64-bit architecture)

### ISSUE THREAD

**Issue by pgkos** [ldc-developers/ldc#2086]
The following example fails to compile with ldc2 on 32-bit (x86) architecture, but on 64-bit (x64) architecture it compiles successfully:
```d
void foo(double a) {
}

int main() {
    // the smallest integer which is not exactly representable as a double:
    foo(9007199254740993L);
    return 0;
}
```
When called with `9007199254740992L`, it succeeds on both architectures.

**Comment by kinke** [comment]
What LDC version and which OS are you using? Compiles fine here on Windows for LDC 1.1.1, 1.2 and current master.

**Comment by kinke** [comment]
DMD 2.074 emits an error for the ...3 case (`function foo (double a) is not callable using argument types (long)`), i.e., it seems to back-cast the double to a long and compare it to the original value.

**Comment by pgkos** [comment]
I'm using Arch Linux (32-bit and 64-bit) and the latest version of ldc from the Arch Linux repositories:
`ldc2 --version` on 64-bit:
```
LDC - the LLVM D compiler (1.2.0):
  based on DMD v2.072.2 and LLVM 4.0.0
  built with DMD64 D Compiler v2.074.0
  Default target: x86_64-unknown-linux-gnu
  Host CPU: core2
```
on 32-bit:
```
LDC - the LLVM D compiler (1.2.0):
  based on DMD v2.072.2 and LLVM 4.0.0
  built with DMD32 D Compiler v2.074.0
  Default target: i686-pc-linux-gnu
  Host CPU: yonah
```

**Comment by kinke** [comment]
This appears to be a general front-end issue: https://github.com/dlang/dmd/pull/6831.

**Comment by kinke** [comment]
Fixed upstream since 2.079.
